src/sentry/tasks/groupowner.py

```diff
@@ -0,0 +1,111 @@
+        can_process = True
+        cache_key = "workflow-owners-ingestion:group-{}".format(event.group_id)
+        owner_data = cache.get(cache_key)
+
+        if owner_data and owner_data["count"] >= PREFERRED_GROUP_OWNERS:
+            # Only process once per OWNER_CACHE_LIFE seconds for groups already populated with owenrs.
+            metrics.incr(
+                "sentry.tasks.process_suspect_commits.skipped", tags={"detail": "too_many_owners"}
+            )
+            can_process = False
+        elif owner_data and owner_data["time"] > timezone.now() - GROUP_PROCESSING_DELAY:
+            # Smaller delay for groups without PREFERRED_GROUP_OWNERS owners yet
+            metrics.incr(
+                "sentry.tasks.process_suspect_commits.skipped", tags={"detail": "group_delay"}
+            )
+            can_process = False
+        else:
+            project = Project.objects.get_from_cache(id=event.project_id)
+            owners = GroupOwner.objects.filter(
+                group_id=event.group_id,
+                project=project,
+                organization_id=project.organization_id,
+                type=GroupOwnerType.SUSPECT_COMMIT.value,
+            )
+            owner_count = owners.count()
+            if owner_count >= PREFERRED_GROUP_OWNERS:
+                # We have enough owners already - so see if any are old.
+                # If so, we can delete it and replace with a fresh one.
+                owners = owners.filter(
+                    date_added__lte=timezone.now() - PREFERRED_GROUP_OWNER_AGE
+                ).order_by("-date_added")
+                if not owners.exists():
+                    metrics.incr(
+                        "sentry.tasks.process_suspect_commits.aborted",
+                        tags={"detail": "maxed_owners_none_old"},
+                    )
+                    can_process = False
+
+            owner_data = {"count": owner_count, "time": timezone.now()}
+            cache.set(cache_key, owner_data, OWNER_CACHE_LIFE)
+
+        if can_process:
+            with metrics.timer("sentry.tasks.process_suspect_commits.process_loop"):
+                metrics.incr("sentry.tasks.process_suspect_commits.calculated")
+                try:
+
+                    with metrics.timer(
+                        "sentry.tasks.process_suspect_commits.get_serialized_event_file_committers"
+                    ):
+                        committers = get_event_file_committers(project, event)
+                    new_owners = []
+                    for committer in committers:
+                        if "id" in committer["author"]:
+                            author_id = committer["author"]["id"]
+                            for commit, score in committer["commits"]:
+                                if score >= MIN_COMMIT_SCORE and not [
+                                    aid for aid, score in new_owners if aid == author_id
+                                ]:
```

Comment:
This is potentially a bit inefficient as new_owners grows larger. You might be better off with a map that just keeps the highest score for each user, then when you iterate over it you just sort `map.values()` and it'll basically be drop-in. I'm ok with doing that as a follow-up.